geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderQueue.java

```diff
@@ -755,12 +755,16 @@ public boolean put(Object object) throws InterruptedException, CacheException {
                 bucketFullPath, brq);
           }
           if (brq != null) {
+            boolean intializingLocked = brq.lockWhenRegionIsInitializing();
```

Comment:
Can we add unit tests to make sure failed Initialization lock is held during put.

Reply:
added.